src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/util/NodeConfigCacheReaderUtil.java

```diff
@@ -64,4 +64,24 @@ public static Long readHeapMaxSizeInBytes(
     }
     return null;
   }
+
+  public static Double readOldGenMaxSizeInBytes(
+      final NodeConfigCache nodeConfigCache, final NodeKey esNode) {
+    try {
+      return nodeConfigCache.get(esNode, ResourceUtil.OLD_GEN_MAX_SIZE);
+    } catch (final IllegalArgumentException e) {
+      LOG.error("Exception while reading heap max size from Node Config Cache", e);
+    }
+    return null;
+  }
+
+  public static Double readYoungGenMaxSizeInBytes(
+      final NodeConfigCache nodeConfigCache, final NodeKey esNode) {
+    try {
+      return nodeConfigCache.get(esNode, ResourceUtil.YOUNG_GEN_MAX_SIZE);
+    } catch (final IllegalArgumentException e) {
```

Comment:
The whole idea of throwing an exception from `nodeConfigCache.get` was to make callers aware of a missing value and **letting them decide** how to handle it. By masking it and returning `null` in a reader util, we are exposing ourselves to future bugs.
I don't understand how this is useful or better, the caller still needs to check for null values - only now the compiler will not ensure that these values are checked. So we run a greater risk of missing these values. That is the whole point of checked exceptions.
This reader util class is good in that you don't have to remember the actual `Resource` values to pass, but it should follow the same API signature. We should not silently eat these exceptions here and make callers handle a different unchecked code path.

PS: I know this is not a new artifact in this PR, it is simply extending existing code faults. I think we should fix it, if the change is big, we can create separate PRs. Either way, at least the new methods can follow the right signature.

Reply:
Also `nodeConfigCache.get` should throw a checked exception, not a RuntimeException (which are for critical/unrecoverable failures)

Reply:
I disagree with your comment on 2 levels

1. The purpose of `NodeConfigCacheReaderUtil` is to eat the exceptions and let consumers gracefully handle nulls as you can see is being done today. This is a valid decision and seems unlikely to lead to future bugs because all consumers today are using this the same way and future consumers are likely to copy the same mechanism.

1. We don't allow storing null values in the NodeConfigCache even if the signature technically allows it. Look at the signature of put(), we can only store actual double values in the cache. We have to use a boxed type for the cache because Java. Since it isn't possible to have null values in the cache, a ReaderUtil returning null will not miss any values and will signal what we want it to signal.

Let me know if you see an issue with my reasoning.

Reply:
I removed the functions I added to this class, my consumers down the line will use the NodeConfigCache directly